components/console-backend-service/internal/authz/auth.go

```diff
@@ -40,11 +44,31 @@ func NewAuthorizer(client authorizationclient.SubjectAccessReviewInterface, cach
 }
 
 // PrepareAttributes prepares attributes for authorization
-func PrepareAttributes(ctx context.Context, u user.Info, attributes gqlschema.ResourceAttributes, obj interface{}) (authorizer.Attributes, error) {
+func PrepareAttributes(ctx context.Context, u user.Info, attributes gqlschema.ResourceAttributes, obj interface{}, client discovery.DiscoveryInterface) (authorizer.Attributes, error) {
```

Comment:
This method has really high cyclomatic complexities.
gocyclo calculated the complexity as **34** points wherein `internal` pkg we have avg 3pkt.

This method has 122 lines. Has a lot of `if, else, switch` which are even nested, check that:

```go
	if attributes.IsChildResolver {
		for i := 0; i < val.NumField(); i++ {
			if attributes.NameArg != nil {
				if nameArg := *attributes.NameArg; fieldName == nameArg {
					if val.Field(i).Kind() != reflect.String {
					}
				}
			}
			if attributes.NamespaceArg != nil {
				if namespaceArg := *attributes.NamespaceArg; fieldName == namespaceArg {
					if val.Field(i).Kind() != reflect.String {
					}
				}
			}
		}
		if attributes.NameArg != nil && name == "" {
		}
		if attributes.NamespaceArg != nil && namespace == "" {
		}
	} else {
```

in one `if` statement u have `for` in which u have `if` in which u have `if` in which u have `if`.

You should avoid to write method like that. You should extract some logic to dedicated function sand the name of the function also needs to be self-describing :)

Reply:
Great points, some of your comments were not related to my contribution (`if in if in if in for in if`), but I totally agree that the function requires a refactor. Never heard about gocyclo, so I'm glad you mentioned it. :)